<?xml version="1.0"?>
<launch>

  <arg name="filter"  				  default="false" />
  <arg name="outer_loop"  			default="false" />
  <arg name="inner_loops"       default="false" /> 
  <arg name="docking_helper"    default="false" /> 

  <group ns="docking">
    
    <rosparam command="load" file="$(find farol_docking)/config/docking_config.yaml"/>

	  <group if="$(arg filter)">
      <node pkg="farol_docking" type="docking_filter" name="filter" respawn="false" output="screen"/>
    </group>

    <group if="$(arg outer_loop)">
      <node pkg="farol_docking" type="outer_loop" name="outer_loop" respawn="false" output="screen"/>
    </group>

    <group if="$(arg inner_loops)">
      <node pkg="farol_docking" type="inner_loops" name="inner_loops" respawn="false" output="screen"/>
    </group>

    <group if="$(arg docking_helper)">
      <node pkg="farol_docking" type="docking_helper.py" name="docking_helper" respawn="false" output="screen"/>
    </group>

  </group>

</launch>
